src/test/ui/impl-trait/must_outlive_least_region_or_bound.stderr

```diff
@@ -50,14 +98,72 @@ LL | fn move_lifetime_into_fn<'a, 'b>(x: &'a u32, y: &'b u32) -> impl Fn(&'a u32
    |                                                 this parameter and the return type are declared with different lifetimes...
 
 error[E0310]: the parameter type `T` may not live long enough
-  --> $DIR/must_outlive_least_region_or_bound.rs:22:51
+  --> $DIR/must_outlive_least_region_or_bound.rs:43:51
    |
 LL | fn ty_param_wont_outlive_static<T:Debug>(x: T) -> impl Debug + 'static {
    |                                 --                ^^^^^^^^^^^^^^^^^^^^ ...so that the type `T` will meet its required lifetime bounds
    |                                 |
    |                                 help: consider adding an explicit lifetime bound...: `T: 'static +`
 
-error: aborting due to 5 previous errors
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/must_outlive_least_region_or_bound.rs:18:50
+   |
+LL | fn elided3(x: &i32) -> Box<dyn Debug> { Box::new(x) }
+   |               ----                               ^ ...is captured here requiring it to live as long as `'static`
+   |               |
+   |               this data with an anonymous lifetime `'_`...
+   |
+help: to permit non-static references in a trait object value, you can add an explicit bound for an anonymous lifetime `'_`
+   |
+LL | fn elided3(x: &i32) -> Box<dyn Debug + '_> { Box::new(x) }
+   |                                      ^^^^
+
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/must_outlive_least_region_or_bound.rs:21:59
+   |
+LL | fn explicit3<'a>(x: &'a i32) -> Box<dyn Debug> { Box::new(x) }
+   |                     -------                               ^ ...is captured here requiring it to live as long as `'static`
+   |                     |
+   |                     this data with lifetime `'a`...
+   |
+help: to permit non-static references in a trait object value, you can add an explicit bound for lifetime `'a`
+   |
+LL | fn explicit3<'a>(x: &'a i32) -> Box<dyn Debug + 'a> { Box::new(x) }
+   |                                               ^^^^
+
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/must_outlive_least_region_or_bound.rs:24:60
+   |
+LL | fn elided4(x: &i32) -> Box<dyn Debug + 'static> { Box::new(x) }
+   |               ----                                         ^ ...is captured here requiring it to live as long as `'static`
+   |               |
+   |               this data with an anonymous lifetime `'_`...
```

Comment:
When we have sentences like this, can we make sure that the first fragment is always above the second? I think users will probably read left-to-right over top-to-bottom, but it wouldn't hurt to make it unambiguous.

Reply:
This is what this code would look like, with more realistic formatting:

<img width="599" alt="Screen Shot 2020-06-01 at 4 17 31 PM" src="https://user-images.githubusercontent.com/1606434/83463632-73b0d200-a423-11ea-9c9e-643675f67e07.png">

Reply:
I see: it's not so much of a problem in general, then. Would it be difficult to alter the behaviour with regards to which labels are pushed down? If we _can_ improve the situation even for more uncommon cases, it seems worth doing, unless it's significant work.